src/context/auth-context.js

```diff
@@ -0,0 +1,54 @@
+import axios from "axios";
+import { useNavigate } from "react-router-dom";
+import { useEffect, createContext, useContext, useReducer } from "react";
+import { authInitialState, authReducer } from "../reducers/reducer";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, authInitialState);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (localStorage.getItem("encodedToken") !== null) {
```

Comment:
You can do this in initial state if possible

Reply:
yeah sure, will do. thanks!